WordPress/src/main/java/org/wordpress/android/ui/mediapicker/MediaPickerFragment.kt

```diff
@@ -45,16 +50,80 @@ import org.wordpress.android.util.image.ImageManager
 import javax.inject.Inject
 
 class MediaPickerFragment : Fragment() {
-    enum class MediaPickerIcon {
+    enum class MediaPickerIconType {
+        ANDROID_CHOOSE_FROM_DEVICE,
         WP_STORIES_CAPTURE;
+
+        companion object {
+            @JvmStatic
+            fun fromNameString(iconTypeName: String): MediaPickerIconType {
+                return values().firstOrNull { it.name == iconTypeName }
+                        ?: throw IllegalArgumentException("MediaPickerIconType not found with name $iconTypeName")
+            }
+        }
+    }
+
+    enum class ChooserContext(
+        val intentAction: String,
+        val requestCode: Int,
+        val title: UiStringRes,
+        val mediaTypeFilter: String
+    ) {
+        PHOTO(ACTION_GET_CONTENT, RequestCodes.PICTURE_LIBRARY, UiStringRes(R.string.pick_photo), "image/*"),
+        VIDEO(ACTION_GET_CONTENT, RequestCodes.VIDEO_LIBRARY, UiStringRes(R.string.pick_video), "video/*"),
+        PHOTO_OR_VIDEO(ACTION_GET_CONTENT, RequestCodes.MEDIA_LIBRARY, UiStringRes(R.string.pick_media), "*/*"),
+        MEDIA_FILE(ACTION_OPEN_DOCUMENT, RequestCodes.FILE_LIBRARY, UiStringRes(R.string.pick_file), "*/*");
+    }
+
+    sealed class MediaPickerAction {
+        data class OpenSystemChooser(
```

Comment:
I like this approach! Very low important - I'd consider using `Picker` instead of `Chooser` here for consistency - we have a Media picker and we'd open System picker here :-)

Reply:
I like it 👍, changed here 4b05415acede9b7e3adf763a87d6b4e823f75315